tests/dragonfly/cluster_test.py

```diff
@@ -284,3 +284,125 @@ async def test_cluster_slot_ownership_changes(df_local_factory):
     assert await c_nodes[0].execute_command("DBSIZE") == 1
     assert (await c_nodes[0].get("KEY0")).decode() == "value"
     assert await c_nodes[1].execute_command("DBSIZE") == 0
+
+
+@dfly_args({"proactor_threads": 4, "cluster_mode": "yes"})
+async def test_cluster_native_client(df_local_factory):
+    # Start and configure cluster with 3 nodes
+    nodes = [
+        df_local_factory.create(port=BASE_PORT+i, admin_port=BASE_PORT+i+1000)
+        for i in range(3)
+    ]
+    df_local_factory.start_all(nodes)
+    c_nodes = [aioredis.Redis(port=node.port) for node in nodes]
+    c_nodes_admin = [aioredis.Redis(port=node.admin_port) for node in nodes]
+    node_ids = await asyncio.gather(*(get_node_id(c) for c in c_nodes_admin))
+
+    config = f"""
+      [
+        {{
+          "slot_ranges": [
+            {{
+              "start": 0,
+              "end": 5000
+            }}
+          ],
+          "master": {{
+            "id": "{node_ids[0]}",
+            "ip": "localhost",
+            "port": {nodes[0].port}
+          }},
+          "replicas": []
+        }},
+        {{
+          "slot_ranges": [
+            {{
+              "start": 5001,
+              "end": 10000
+            }}
+          ],
+          "master": {{
+            "id": "{node_ids[1]}",
+            "ip": "localhost",
+            "port": {nodes[1].port}
+          }},
+          "replicas": []
+        }},
+        {{
+          "slot_ranges": [
+            {{
+              "start": 10001,
+              "end": 16383
+            }}
+          ],
+          "master": {{
+            "id": "{node_ids[2]}",
+            "ip": "localhost",
+            "port": {nodes[2].port}
+          }},
+          "replicas": []
+        }}
+      ]
+    """
+    await push_config(config, c_nodes_admin)
+
+    client = redis.RedisCluster(decode_responses=True, host="localhost", port=nodes[0].port)
+
+    for i in range(10_000):
+        key = 'key' + str(i)
+        assert client.set(key, 'value') == True
+        assert client.get(key) == 'value'
```

Comment:
Small note: you're using the sync client inside an async function

Reply:
Thanks, that's a good catch!
I'm new to Python async (and also to Python in general).
My goal here is to use `RedisCluster`, which seems to be a sync library. Is there a way for me to still use the other async tools we have here, like `push_config()` and `get_node_id()`?
Alternatively I guess I could use `run_in_executor()` (from [here](https://stackoverflow.com/a/54685298)), do you think that's advisable?

Reply:
You can use the async `aioredis.RedisCluster` (which is now just a re-export of the official redis.asyncio). It should have the same interface

If there were only the sync version, wouldn't have caused any trouble either way, as you only perform sequential operations and don't have any tasks running

Reply:
That worked, thanks!